File: Algorithms/Javascript/binarySearchRecursive.js
const binarySearch = (arr, left, right, target) => {
    if (left > right) {
        return -1;
    }

    const mid = Math.floor((left + right) / 2);

    if (arr[mid] === target) {
        return mid;
    }

    if (arr[mid] > target) {
        return binarySearch(arr, left, mid - 1, target);
    } else {
        return binarySearch(arr, mid + 1, right, target);
    }
};

const arr = [1, 2, 3, 4, 5, 6, 7, 8];
const target = 8;

const index = binarySearch(arr, 0, arr.length - 1, target);

if (index == -1) {
    console.log("Not found");
} else {
    console.log("Found at index: " + index);
}

File: Algorithms/C/binary_search_recursive.c
#include <stdio.h>

void binary_search(const int arr[], int left, int right, int target)
{
    if (left > right)
    {
        printf("Not found %d\n", target);
        return;
    }

    int mid = (left + right) / 2;
    if (arr[mid] == target)
    {
        printf("Found %d at index %d\n", target, mid);
        return;
    }
    if (arr[mid] < target)
    {
        binary_search(arr, mid + 1, right, target);
    }
    else
    {
        binary_search(arr, left, mid - 1, target);
    }
}

int main()
{
    int arr[] = {1, 2, 3, 4, 5, 6};
    int len = sizeof(arr) / sizeof(arr[0]);
    binary_search(arr, 0, len - 1, 3);
    binary_search(arr, 0, len - 1, 7);
    return 0;
}
